### PR TITLE
CDC: Implement Initializer of Endpoint

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

###  What have you changed?

This PR implements the `Initializer` that's used by CDC's `Endpoint` to initialize `Delegate`s.  `Endpoint` and `Delegate` are either unimplemented yet on master, so `Initializer` is actually not used for now.

###  What is the type of the changes?

- New feature (a change which adds functionality)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by MyonKeminta** [comment]
@overvenus @5kbpers PTAL thanks

File: components/cdc/src/endpoint.rs [hunk 1]
```diff
@@ -120,25 +142,341 @@ impl Endpoint {
...
+                // Time range: (checkpoint_ts, current]
+                let current = TimeStamp::max();
+                let mut scanner = ScannerBuilder::new(snap, current, false)
+                    .range(None, None)
+                    .build_delta_scanner(checkpoint_ts)
+                    .unwrap();
+                let mut done = false;
+                while !done {
+                    let entries =
+                        match Self::scan_batch(&mut scanner, batch_size, resolver.as_mut()) {
```
Comment (overvenus) [inline components/cdc/src/endpoint.rs]: Can we use `TxnEntryScanner::scan_entries`?

Comment (MyonKeminta) [reply components/cdc/src/endpoint.rs]: The current implementation in CDC branch needs `Vec<Option<TxnEntry>>` here, and a `None` is used to indicate that scanning has finished. If we use scan_entries here it seems this behavior is better to be changed. Is that ok?

Comment (overvenus) [reply components/cdc/src/endpoint.rs]: An empty `EntryBatch` means finished. I don't think there is a big difference in behavior.

Comment (MyonKeminta) [reply components/cdc/src/endpoint.rs]: ok

Comment (MyonKeminta) [reply components/cdc/src/endpoint.rs]: PTAL again. I kept the scan_batch function to group scanning and resolver building together

File: components/cdc/src/endpoint.rs [hunk 2]
```diff
@@ -120,25 +142,331 @@ impl Endpoint {
...
+            }))
+            .unwrap();
+    }
+
+    fn scan_batch<S: Snapshot>(
+        scanner: &mut DeltaScanner<S>,
+        batch_size: usize,
+        resolver: Option<&mut Resolver>,
+    ) -> Result<EntryBatch> {
+        let mut entry_batch = EntryBatch::with_capacity(batch_size);
```
Comment (overvenus) [inline components/cdc/src/endpoint.rs]: It allocates every time we call this function, could you reuse the buffer?

Comment (MyonKeminta) [reply components/cdc/src/endpoint.rs]: oh, good idea

Comment (MyonKeminta) [reply components/cdc/src/endpoint.rs]: The batch is then sent back to the scheduler at line 252 which makes it difficult to reuse in the current architecture.

**Comment by MyonKeminta** [comment]
@5kbpers PTAL again thx!

**Pull request merged**